dataframe-csv/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/DelimCsvTsvTests.kt

```diff
@@ -0,0 +1,657 @@
+        df.columnsCount() shouldBe 2
+        df["first"].type() shouldBe typeOf<Int>()
+        df["second"].allNulls() shouldBe true
+        df["second"].type() shouldBe typeOf<String?>()
+    }
+
+    @Test
+    fun write() {
+        val df = dataFrameOf("col1", "col2")(
+            1, null,
+            2, null,
+        ).convert("col2").toStr()
+
+        val str = StringWriter()
+        df.writeCsv(str)
+
+        val res = DataFrame.readCsvStr(str.buffer.toString())
+
+        res shouldBe df
+    }
+
+    @Test
+    fun readCsv() {
+        val df = DataFrame.read(simpleCsv)
+
+        df.columnsCount() shouldBe 11
+        df.rowsCount() shouldBe 5
+        df.columnNames()[5] shouldBe "duplicate1"
+        df.columnNames()[6] shouldBe "duplicate11"
+        df["duplicate1"].type() shouldBe typeOf<Char?>()
+        df["double"].type() shouldBe typeOf<Double?>()
+        df["number"].type() shouldBe typeOf<Double>()
+        df["time"].type() shouldBe typeOf<LocalDateTime>()
+
+        df.print(columnTypes = true, borders = true, title = true)
+    }
+
+    @Test
+    fun `read ZIP Csv`() {
+        val df = DataFrame.readCsv(simpleCsvZip)
+
+        df.columnsCount() shouldBe 11
+        df.rowsCount() shouldBe 5
+        df.columnNames()[5] shouldBe "duplicate1"
+        df.columnNames()[6] shouldBe "duplicate11"
+        df["duplicate1"].type() shouldBe typeOf<Char?>()
+        df["double"].type() shouldBe typeOf<Double?>()
+        df["number"].type() shouldBe typeOf<Double>()
+        df["time"].type() shouldBe typeOf<LocalDateTime>()
+
+        df.print(columnTypes = true, borders = true, title = true)
+    }
+
+    @Test
+    fun `read GZ Csv`() {
+        val df = DataFrame.readCsv(simpleCsvGz)
+
+        df.columnsCount() shouldBe 11
+        df.rowsCount() shouldBe 5
+        df.columnNames()[5] shouldBe "duplicate1"
+        df.columnNames()[6] shouldBe "duplicate11"
+        df["duplicate1"].type() shouldBe typeOf<Char?>()
+        df["double"].type() shouldBe typeOf<Double?>()
+        df["number"].type() shouldBe typeOf<Double>()
+        df["time"].type() shouldBe typeOf<LocalDateTime>()
+
+        df.print(columnTypes = true, borders = true, title = true)
+    }
+
+    @Test
+    fun `read custom compression Csv`() {
+        val df = DataFrame.readCsv(
+            simpleCsvGz,
+            compression = Compression.Custom { GZIPInputStream(it) },
+        )
+
+        df.columnsCount() shouldBe 11
+        df.rowsCount() shouldBe 5
+        df.columnNames()[5] shouldBe "duplicate1"
+        df.columnNames()[6] shouldBe "duplicate11"
+        df["duplicate1"].type() shouldBe typeOf<Char?>()
+        df["double"].type() shouldBe typeOf<Double?>()
+        df["number"].type() shouldBe typeOf<Double>()
+        df["time"].type() shouldBe typeOf<LocalDateTime>()
+
+        df.print(columnTypes = true, borders = true, title = true)
+    }
+
+    @Test
+    fun `read 2 compressed Csv`() {
```

Comment:
Do you have a test for reading something zipped, but that is not CSV or has an extension, but in reality has a weird format internally